cmd/ausoceantv/main.go

```diff
@@ -120,6 +136,75 @@ func (svc *service) setup(ctx context.Context) {
 	log.Printf("set up datastore")
 }
 
+// handleCreatePaymentIntent handles requests to /stripe/create-payment-intent.
+func handleCreatePaymentIntent(w http.ResponseWriter, r *http.Request) {
+	app.logRequest(r)
+
+	// Enable Cross-Origin Scripting from Vite.
+	enableCors(&w)
+	if r.Method == http.MethodOptions {
+		w.WriteHeader(200)
+		return
+	} else if r.Method != "POST" {
+		http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+		return
+	}
+
+	// TODO: Get product details.
+	//	description := product.description
+	//	price := calculatePrice(product)
+
+	// Enable auto payment method for better conversions.
+	autoPaymentMethodEnabled := true
+
+	// Create a PaymentIntent with amount and currency.
+	params := &stripe.PaymentIntentParams{
+		Amount:                  stripe.Int64(1099),
+		Currency:                stripe.String(string(stripe.CurrencyAUD)),
+		AutomaticPaymentMethods: &stripe.PaymentIntentAutomaticPaymentMethodsParams{Enabled: &autoPaymentMethodEnabled},
+	}
+
+	// NOTE: DO NOT LOG PAYMENT INTENT.
+	pi, err := paymentintent.New(params)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("Error creating new Stripe payment intent: %v", err)
+		return
+	}
+
+	writeJSON(w, struct {
+		ClientSecret string `json:"clientSecret"`
+		// DpmCheckerLink string `json:"dpmCheckerLink"` <-- Can be used for debugging of the integration.
+	}{
+		ClientSecret: pi.ClientSecret,
+		// [DEV]: For demo purposes only, you should avoid exposing the PaymentIntent ID in the client-side code.
+		// DpmCheckerLink: fmt.Sprintf("https://dashboard.stripe.com/settings/payment_methods/review?transaction_id=%s", pi.ID),
+	})
+}
+
+// enableCors allows the vite server to read responses from this webserver.
+func enableCors(w *http.ResponseWriter) {
+	(*w).Header().Set("Access-Control-Allow-Origin", "http://localhost:5173")
```

Comment:
This should also reflect the correct address of vite, which I think may be :4173 in production?

Reply:
no need to dereference in Go FYI, it happens for you :)